src/hooks/useTokenBalance.ts

```diff
@@ -61,4 +63,24 @@ export const useBurnedBalance = (tokenAddress: string) => {
   return balance
 }
 
+export const useGetBnbBalance = () => {
+  const [balance, setBalance] = useState(new BigNumber(0))
+  const { account } = useWeb3React()
+  const { lastUpdated, setLastUpdated } = useLastUpdated()
+
+  useEffect(() => {
+    const fetchBalance = async () => {
+      const web3 = getWeb3NoAccount()
+      const walletBalance = await web3.eth.getBalance(account)
+      setBalance(new BigNumber(walletBalance))
+    }
+
+    if (account) {
+      fetchBalance()
+    }
+  }, [account, lastUpdated, setBalance])
+
+  return { balance, refresh: setLastUpdated }
```

Comment:
`refresh` is so the consumer of the hook can re-fetch the balance at their leisure

Reply:
This is neat. I actually hadn't seen you add that `useLastUpdated` hook - it should be pretty useful! I found myself toggling a state boolean at points in the trading comp to effectively do the same thing.